### DISCUSSION TITLE
Problem with RichText field when template displays

### DISCUSSION THREAD

**Discussion by vanevalverdes** [pocketbase/pocketbase#3476] [Q&A] [answered]
Hello, I am trying to display a content of the Rich Editor field type. But when displayed in the template, it is not interpreted as an html and displays tags like <p> and &nbsp;
I tried using {{.description | html}} and nothing solves it. How can I solve that?
Thank you,

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. I'll assume that you've passed a Record model as `render` data to the template.

In this case public struct fields and methods should be accessed with uppercase similar as in Go (because we use the Go templates).

Or in other words, you can try with something like `{{.Get("description")}}`.

If you pass as `render` data `record.publicExport()` then you should be able to access the keys with lowercase because in this case they `.` object will be a regular map and `description` will be a key in it.

If the above doesn't help, please provide a more complete code sample.

**Reply by ganigeorgiev** [reply, marked as answer]
~_On second read, if your problem is actually that by default the content of your field is not displayed as raw HTML, then you can use the `{{html .description}}` function (for all available builtin functions you can check https://pkg.go.dev/text/template#hdr-Functions)._~

Ignore the above. For security reasons the `html/template` Go package will escape the content by default and it seems to require that raw HTML strings to be presented as [`template.HTML` type](https://pkg.go.dev/html/template#HTML) but currently there is no such option in the JS `pb_hooks`. I'll experiment a little more sometime later and will see if it can be added as a global template function or at least a JSVM helper.

**Reply by vanevalverdes** [reply]
Oh, thanks! That's is the problema. Because I used {{html .record.description}} for displays this field in the template, with JS pb_hooks. Then, if I develop with the extends with GO option, I can displays like raw HTML in the template?

Thanks again

**Reply by ganigeorgiev** [reply]
If you use Go, casting the `description` value with `template.HTML(description)` should work, for example:

```go
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
    // this is safe to be used by multiple goroutines
    // (it acts as store for the parsed templates)
    registry := template.NewRegistry()

    e.Router.GET("/example", func(c echo.Context) error {
        description := "..."

        html, err := registry.LoadFiles(
            "views/layout.html",
            "views/example.html",
        ).Render(map[string]any{
            "description": template.HTML(description),
        })

        if err != nil {
            // or redirect to a dedicated 404 HTML page
            return apis.NewNotFoundError("", err)
        }

        return c.HTML(http.StatusOK, html)
    })

    return nil
})
```

But I'll consider tomorrow to add support for raw HTML output in the JS `pb_hooks` and eventually will push a minor v0.18.10 release.

**Reply by vanevalverdes** [reply]
For the moment, I used a JS function in the template, to put my raw HTML content in to a variable and insert in a div... And this works.

`function mostrarHTML() {
  var div = document.getElementById("desc");
  var htmlString = "{{html .record.description}}";
  div.insertAdjacentHTML("beforeend", htmlString);
}
window.onload = mostrarHTML;`

**Reply by ganigeorgiev** [reply]
A `raw` function was added in [v0.18.10 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.18.10), aka. doing something like `{{.description | raw}}` should output the description as it is without escaping the HTML entities.

**Reply by vanevalverdes** [reply]
Oh! thanks for this information. It's very useful for us. :)
